internal/clientconn/conninfo/conn_info.go

```diff
@@ -37,6 +37,8 @@ type ConnInfo struct {
 	rw       sync.RWMutex
 	username string
 	password string
+
+	ClientMetadataPresence bool
```

Comment:
```suggestion
	clientMetadataPresence bool
```

To prevent direct access and use the same ["mutex hat"](https://dmitri.shuralyov.com/idiomatic-go#mutex-hat)